scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+
+    echo "huchen: occupiedIPList is ${occupiedIPList}"
+    IFS=$'\n'
+    for occupiedIPWithMask in ${occupiedIPList}
+    do
+        echo "huchen: occupiedIPWithMask is ${occupiedIPWithMask}"
+        occupiedIP=`echo $occupiedIPWithMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+        echo "huchen: ip is ${ip}, occupiedIP is ${occupiedIP}"
+        if [ "${ip}" == "${occupiedIP}" ]; then
+             return 1
+        fi
+    done
+    echo "${ip} is not occupied"
+    return 0
+}
+
+
+function num2IP() {
+    num=$1
+    a=$((num>>24))
+    b=$((num>>16&0xff))
+    c=$((num>>8&0xff))
+    d=$((num&0xff))
+    echo "$a.$b.$c.$d"
+}  
+
+function binary2IP() {
+    num=`echo "ibase=2;  $1" | bc`
+    returnValue=$(num2IP $num)
+    echo ${returnValue}
+}
+
+
+function calcControlPlaneIP() {
+    subNet=`docker network inspect kind | jq -r 'map(.IPAM.Config[].Subnet) []'`
+    occupiedIPList=`docker network inspect kind | jq -r 'map(.Containers[].IPv4Address) []'`
+
+    for line in ${subNet}
+    do
+        echo ${line} | grep ":"
+        if [ $? -ne 0 ]; then
+            IPMask=${line}
+            break
+        fi
+    done
+
+    echo "IPMask is ${subNet}"
+    echo "occupiedIPList:${occupiedIPList}"
+
+    ip=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\1/p'`
+    ipSubNetBit=`echo $IPMask | sed -En 's/^(.*)\/([0-9]{1,2})/\2/p'`
+    ipSubHostBit=$[32-${ipSubNetBit}]
+
+    IFS=.
+    hexIP=`for str in ${ip}; do printf "%02X" $str; done`
+    binaryIP=`echo "ibase=16; obase=2; ${hexIP}" | bc`
+
+    ipSubNet=`echo ${binaryIP:0:${ipSubNetBit}}`
+
+    full0="00000000000000000000000000000010"
+    full1="11111111111111111111111111111110"
+
+    minIPBinary=${ipSubNet}`echo ${full0:${ipSubNetBit}}`
+    maxIPBinary=${ipSubNet}`echo ${full1:${ipSubNetBit}}`
+
+    minIPint=`echo "ibase=2; ${minIPBinary}"|bc`
+    maxIPint=`echo "ibase=2; ${maxIPBinary}"|bc`
+
+
+    for((i=${minIPint};i<=${maxIPint};i++));  
+    do
+        ip=$(num2IP ${i})
+
+        isIPOccupied "${ip}" "${occupiedIPList}"
+        if [ $? == 0 ]; then
+            CONTROL_PLANE_ENDPOINT_IP=${ip}
+            echo "Available IP is ${CONTROL_PLANE_ENDPOINT_IP}"
+            return
+        fi
+    done
+}
+
+function configClusterctl() {
+    # Write clusterctl.yaml
+    writeByoh=0
+    writeCertManager=0
+    clusterCtlYamlFile="${HOME}/.cluster-api/clusterctl.yaml"
+    if [ ! -f "${clusterCtlYamlFile}" ]; then
+        writeByoh=1
+        writeCertManager=1
+        touch ${clusterCtlYamlFile}
+    else
+        grep -q "byoh" ${clusterCtlYamlFile}
+        if [ $? -ne 0 ] ; then
+            writeByoh=1
+        fi
+        grep -q "cert-manager" ${clusterCtlYamlFile}
+        if [ $? -ne 0 ] ; then
+            writeCertManager=1
+        fi
+    fi 
+
+    # if cert-manager missing,it report error: Error: action failed after 10 attempts: failed to get cert-manager object /, Kind=, /: Object 'Kind' is missing in 'unstructured object has no kind'
```

Comment:
You are probably getting this error because of an older version of clusterctl. This was fixed in `v1.0.4` or directly use the latest version of clusterctl

Reply:
ok. Will try the latest one to see if it can fix this issue.